lazyflow/operators/opRelabelConsecutive.py

```diff
@@ -1,26 +1,506 @@
 
 logger = logging.getLogger(__name__)
 
+RoiTuple = Tuple[Tuple[int, ...], Tuple[int, ...]]
+Roi5D = Tuple[Tuple[int, int, int, int, int], Tuple[int, int, int, int, int]]
+
 
 class OpRelabelConsecutive(Operator):
+    """
+    Operator for relabeling label images in a consecutive manner
+
+    This operator sets up the inputs for OpRelabelConsecutive5D:
+    * `BlockShape` is set up in to whole time slices, no spatial subblocks
+      for valid relabelings.
+    * `dtype` of the `Input` is checked (see `supportedDtypes`) and in case of
+      `uint16` changed to `uint32` to comply with vigra requirements. `uint16` is
+       what we have seen with cellpose data.
+    * `Input` is always reordered to `tzyxc`
+    """
+
     Input = InputSlot()
-    StartLabel = InputSlot(value=0)
-    Output = OutputSlot()
+    StartLabel = InputSlot(value=1)
+
+    CachedOutput = OutputSlot()
+    RelabelDict = OutputSlot()
+
+    CleanBlocks = OutputSlot()
+    CompressionEnabled = InputSlot(value=False)
+
+    SerializationInput = InputSlot(optional=True)
+    SerializationOutput = OutputSlot()
+
+    # total dummy slot, there because OpObjectExtraction expects this interface
+    BypassModeEnabled = InputSlot(value=False)
+
+    supportedDtypes = [numpy.uint8, numpy.uint16, numpy.uint32, numpy.uint64]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._opReorder5D = OpReorderAxes(parent=self, Input=self.Input, AxisOrder="tzyxc")
+        self._opDtypeConvert = OpPixelOperator(parent=self, Input=self._opReorder5D.Output, Function=lambda x: x)
+
+        self._opRelabelConsecutive = OpRelabelConsecutive5D(
+            parent=self,
+            Input=self._opDtypeConvert.Output,
+            StartLabel=self.StartLabel,
+            CompressionEnabled=self.CompressionEnabled,
+            SerializationInput=self.SerializationInput,
+        )
+        self._opReorderToInputOrder = OpReorderAxes(parent=self)
+        self.CachedOutput.connect(self._opReorderToInputOrder.Output)
+        self.RelabelDict.connect(self._opRelabelConsecutive.RelabelDict)
+        self.CleanBlocks.connect(self._opRelabelConsecutive.CleanBlocks)
+        self.SerializationOutput.connect(self._opRelabelConsecutive.SerializationOutput)
 
     def setupOutputs(self):
-        self.Output.meta.assignFrom(self.Input.meta)
+        tagged_input = self._opReorder5D.Output.meta.getTaggedShape()
+        tagged_blockshape = {k: v for k, v in tagged_input.items()}
+        tagged_blockshape["t"] = 1
+
+        input_dtype = self.Input.meta.dtype
+
+        if input_dtype == numpy.uint16:
+            self._opDtypeConvert.Function.setValue(lambda x: x.astype("uint32"))
+        else:
+            self._opDtypeConvert.Function.setValue(lambda x: x)
+
+        self._opReorderToInputOrder.AxisOrder.setValue("".join(self.Input.meta.getAxisKeys()))
+        self._opReorderToInputOrder.Input.connect(self._opRelabelConsecutive.Output)
```

Comment:
We need the input to be ready in order to infer shape/axisorder. So `setupOutput` ensures this is done only in a state when we have that info. Because we might not now the input metadata yet. The op can happily be constructed without the input being ready.

Reply:
Bad github interface, I was specifically referring to `Input.connect()`, which shouldn't need Input to be ready. Is there a specific reason Input shouldn't be connected before AxisOrder has a value?

Reply:
yeah, the reorder op has a default value for the axisorder - this messes up things.